beacon-chain/sync/slot_aware_cache.go

```diff
@@ -0,0 +1,96 @@
+package sync
+
+import (
+	"sort"
+	"sync"
+
+	"github.com/OffchainLabs/prysm/v6/consensus-types/primitives"
+	lru "github.com/hashicorp/golang-lru"
+)
+
+const maxSlots = 1000 // Maximum number of slots to track before pruning oldest
+
+// slotAwareCache is a cache that tracks which keys belong to which slot
+// to enable slot-based pruning when blocks are finalized.
+type slotAwareCache struct {
+	cache      *lru.Cache
+	slotToKeys map[primitives.Slot][]string
+	mu         sync.RWMutex
+}
+
+// newSlotAwareCache creates a new slot-aware cache with the given size.
+func newSlotAwareCache(size int) *slotAwareCache {
+	cache, _ := lru.New(size)
+	return &slotAwareCache{
+		cache:      cache,
+		slotToKeys: make(map[primitives.Slot][]string),
+	}
+}
+
+// Get retrieves a value from the cache.
+func (c *slotAwareCache) Get(key string) (interface{}, bool) {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+	return c.cache.Get(key)
+}
+
+// Add adds a value to the cache associated with a specific slot.
+func (c *slotAwareCache) Add(slot primitives.Slot, key string, value interface{}) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	// Add to cache
+	c.cache.Add(key, value)
+
+	// Track slot association
+	c.slotToKeys[slot] = append(c.slotToKeys[slot], key)
+
+	// If we exceed maxSlots, prune the oldest slots
+	if len(c.slotToKeys) > maxSlots {
+		c.pruneOldestSlots()
+	}
```

Comment:
```suggestion
	c.pruneOldestSlots()
```

`pruneOldestSlots` is only called by `Add`, which already check if `len(c.slotToKeys) > maxSlots`.
==> We can skip one check.